deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/onnx/ONNXHub.scala

```diff
@@ -0,0 +1,229 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.onnx
+
+import com.microsoft.azure.synapse.ml.core.env.FileUtilities
+import com.microsoft.azure.synapse.ml.core.env.StreamUtilities.using
+import com.microsoft.azure.synapse.ml.core.utils.FaultToleranceUtils
+import org.apache.commons.codec.digest.DigestUtils
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.hadoop.io.{IOUtils => HUtils}
+import org.apache.spark.SparkContext
+import org.apache.spark.internal.Logging
+import spray.json.{DefaultJsonProtocol, RootJsonFormat, _}
+
+import java.io.BufferedInputStream
+import java.net.URL
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.Duration
+
+case class ONNXShape(name: String, shape: Seq[Either[Option[String], Int]], `type`: Option[String])
+
+case class ONNXIOPorts(inputs: Seq[ONNXShape], outputs: Seq[ONNXShape])
+
+case class ONNXExtraPorts(features: Seq[ONNXShape])
+
+case class ONNXMetadata(modelSha: Option[String],
+                        modelBytes: Option[Long],
+                        tags: Option[Seq[String]],
+                        ioPorts: Option[ONNXIOPorts],
+                        extraPorts: Option[ONNXExtraPorts],
+                        modelWithDataPath: Option[String],
+                        modelWithDataSha: Option[String],
+                        modelWithDataBytes: Option[Long])
+
+case class ONNXModelInfo(model: String,
+                         modelPath: String,
+                         onnxVersion: String,
+                         opsetVersion: Int,
+                         metadata: ONNXMetadata)
+
+object ONNXHubJsonProtocol extends DefaultJsonProtocol {
+  implicit val ShapeFormat: RootJsonFormat[ONNXShape] = jsonFormat3(ONNXShape.apply)
+
+  implicit val IoFormat: RootJsonFormat[ONNXIOPorts] = jsonFormat2(ONNXIOPorts.apply)
+
+  implicit val ExtraFormat: RootJsonFormat[ONNXExtraPorts] = jsonFormat1(ONNXExtraPorts.apply)
+
+  implicit val MetaFormat: RootJsonFormat[ONNXMetadata] = jsonFormat(
+    ONNXMetadata.apply,
+    "model_sha",
+    "model_bytes",
+    "tags",
+    "io_ports",
+    "extra_ports",
+    "model_with_data_path",
+    "model_with_data_sha",
+    "model_with_data_bytes")
+
+  implicit val InfoFormat: RootJsonFormat[ONNXModelInfo] = jsonFormat(
+    ONNXModelInfo.apply,
+    "model",
+    "model_path",
+    "onnx_version",
+    "opset_version",
+    "metadata"
+  )
+}
+
+object ONNXHub {
+  val DefaultRepo: String = "onnx/models:5bb96f84527f6c0cd437f36ad4b0bb7daf5ef830"
+  val AuthenticatedRepo: (String, String, String) = ("onnx", "models", "5bb96f84527f6c0cd437f36ad4b0bb7daf5ef830")
```

Comment:
might want to set to master so that it has the same semantics as python

Reply:
Not sure what you mean, can you give an example?  This is code you wrote and to be honest I didn't even look at it other than to fix spacing. :)